script/testScripts/gov/GenerateProposal/GenerateModifyParametersProposal.s.sol

```diff
@@ -0,0 +1,108 @@
+
+  string public objectKey = 'MODIFY_PARAMS_OBJECT_KEY';
+
+  address[] internal _targets;
+  string[] internal _datas;
+  string[] internal _params;
+  string[] internal _dataTypes;
+
+  address internal _governanceAddress;
+  string internal _description;
+
+  function _loadBaseData(string memory json) internal override {
+    _description = json.readString(string(abi.encodePacked('.description')));
+    _governanceAddress = json.readAddress(string(abi.encodePacked('.odGovernor')));
+    uint256 len = json.readUint(string(abi.encodePacked('.numberOfModifications')));
+
+    for (uint256 i; i < len; i++) {
+      string memory index = Strings.toString(i);
+      address target = json.readAddress(string(abi.encodePacked('.modifyObjects[', index, '].target')));
+      string memory param = json.readString(string(abi.encodePacked('.modifyObjects[', index, '].param')));
+      string memory dataType = json.readString(string(abi.encodePacked('.modifyObjects[', index, '].type')));
+      string memory data = json.readString(string(abi.encodePacked('.modifyObjects[', index, '].data')));
+      _targets.push(target);
+      _params.push(param);
+      _dataTypes.push(dataType);
+      _datas.push(data);
+    }
+  }
+
+  function _generateProposal() internal override {
+    ODGovernor gov = ODGovernor(payable(_governanceAddress));
+
+    require(
+      _params.length == _dataTypes.length && _dataTypes.length == _targets.length, 'Modify Parameters: Length Mismatch'
+    );
+
+    uint256 len = _params.length;
+
+    address[] memory targets = new address[](len);
+    uint256[] memory values = new uint256[](len);
+    bytes[] memory calldatas = new bytes[](len);
+
+    for (uint256 i; i < len; i++) {
+      targets[i] = _targets[i];
+      values[i] = 0;
+      calldatas[i] = _readData(_dataTypes[i], _params[i], _datas[i]);
+    }
+
+    bytes32 descriptionHash = keccak256(bytes(_description));
+
+    vm.startBroadcast(privateKey);
+
+    // Propose the action to add the collateral type
+    uint256 proposalId = gov.hashProposal(targets, values, calldatas, descriptionHash);
+    string memory stringProposalId = vm.toString(proposalId / 10 ** 69);
+
+    {
+      // Build the JSON output
+      string memory builtProp =
+        _buildProposalParamsJSON(proposalId, objectKey, targets, values, calldatas, _description, descriptionHash);
+      vm.writeJson(
+        builtProp, string.concat('./gov-output/', network, '/', stringProposalId, '-modifyParameters-proposal.json')
+      );
+    }
+
+    vm.stopBroadcast();
+  }
+
+  function _readData(
+    string memory dataType,
+    string memory param,
+    string memory dataString
+  ) internal pure returns (bytes memory dataOutput) {
+    bytes32 typeHash = keccak256(abi.encode(dataType));
+    bytes4 selector = IModifiable.modifyParameters.selector;
+    // keccak256(abi.encode('uint256'))
+    if (typeHash == 0x30f3243bbfedd88bc30751e3112f67415273ecd7b3860e727409d9e0be640fe7) {
```

Comment:
rather than write `abi.encodeWithSelector...` for each type, is it possible to just save the parsed data to a variable eg `parsed = vm.parseUint(dataString)`

Also, can we make it easier to read the conditional statement, maybe by comparing the string `uint256` directly?

Reply:
I have to encode it straight into the calldata so I can return the bytes.
comparing strings in solidity is a bit iffy, comparing bytes32 is better.  I can go to calculating the bytes32 each time.  I was just trying to make it more efficient by hardcoding it.  however it doesn't really matter for the scripts since they're not on chain.